samples/ChangefeedMigrationSample/Program.cs

```diff
@@ -0,0 +1,222 @@
+    {
+        [JsonProperty("id")]
+        public string Id;
+
+        [JsonProperty("deviceId")]
+        public string DeviceId;
+
+        [JsonConverter(typeof(IsoDateTimeConverter))]
+        [JsonProperty("readingTime")]
+        public DateTime ReadingTime;
+
+        [JsonProperty("metricType")]
+        public string MetricType;
+
+        [JsonProperty("unit")]
+        public string Unit;
+
+        [JsonProperty("metricValue")]
+        public double MetricValue;
+    }
+
+    class DocumentFeedObserver : IChangeFeedObserver
+    {
+        private static int s_totalDocs = 0;
+
+        public Task OpenAsync(ChangeFeedObserverContext context)
+        {
+            Console.WriteLine("Worker opened, {0}", context.PartitionKeyRangeId);
+            return Task.CompletedTask;  // Requires targeting .NET 4.6+.
+        }
+
+        public Task CloseAsync(ChangeFeedObserverContext context, ChangeFeedObserverCloseReason reason)
+        {
+            Console.WriteLine("Worker closed, {0}", context.PartitionKeyRangeId);
+            return Task.CompletedTask;
+        }
+
+        public Task ProcessChangesAsync(ChangeFeedObserverContext context, IReadOnlyList<Document> docs)
+        {
+            DocumentCollectionInfo destCollectionLocation = new DocumentCollectionInfo
+            {
+                Uri = new Uri("httpsURI"),
+                MasterKey = "PrimaryKey",
+                DatabaseName = "DestDBname",
+                CollectionName = "DestCollName"
+            };
+
+            DocumentClient newClient = new DocumentClient(destCollectionLocation.Uri, destCollectionLocation.MasterKey);
+
+            newClient.CreateDatabaseIfNotExistsAsync(new Database { Id = destCollectionLocation.DatabaseName });
+
+            // create dest collection if it does not exist
+            DocumentCollection destCollection = new DocumentCollection();
+            destCollection.Id = destCollectionLocation.CollectionName;
+
+            //destCollection.PartitionKey.Paths.Add("add partition key if applicable");
+
+            newClient.CreateDocumentCollectionIfNotExistsAsync(
+                UriFactory.CreateDatabaseUri(destCollectionLocation.DatabaseName),
+                destCollection,
+                new RequestOptions { OfferThroughput = 500 });
+
+            Console.WriteLine("Change feed: total {0} doc(s)", Interlocked.Add(ref s_totalDocs, docs.Count));
+            foreach (Document doc in docs)
+            {
+                Console.WriteLine(doc.ToString());
+                newClient.UpsertDocumentAsync(
+                    UriFactory.CreateDocumentCollectionUri(destCollectionLocation.DatabaseName, destCollectionLocation.CollectionName),
+                    doc);
+            }
+            return Task.CompletedTask;
+        }
+    }
+
+    class Program
+    {
+        // Modify EndPointUrl and PrimaryKey to connect to your own subscription
+        private string uri = "https://URI";
```

Comment:
I would add a second Uri and a second Database variables for the Leases and comment that the Leases can be stored on a different account. A user might have a Partitioned and globally distributed account for his data but a single region account for his Leases because he doesn't want to globally replicate the leases.